packages/dd-trace/src/telemetry/index.js

```diff
@@ -1,393 +1,40 @@
-  telemetryLogger.start(config)
-
-  sendData(config, application, host, 'app-started', appStarted(config))
-
-  if (integrations.length > 0) {
-    sendData(config, application, host, 'app-integrations-change',
-      { integrations }, updateRetryData)
-  }
-
-  heartbeat(config, application, host)
-
-  extendedHeartbeat(config)
-
-  process.on('beforeExit', onBeforeExit)
-  telemetryStartChannel.publish(getTelemetryData())
-}
-
-function stop () {
-  if (!config) {
-    return
-  }
-  clearInterval(extendedInterval)
-  clearTimeout(heartbeatTimeout)
-  process.removeListener('beforeExit', onBeforeExit)
-
-  telemetryStopChannel.publish(getTelemetryData())
-
-  config = undefined
-}
-
-function updateIntegrations () {
-  if (!config || !config.telemetry.enabled) {
-    return
-  }
-  const integrations = getIntegrations()
-  if (integrations.length === 0) {
-    return
-  }
-
-  const { reqType, payload } = createPayload('app-integrations-change', { integrations })
-
-  sendData(config, application, host, reqType, payload, updateRetryData)
-}
-
-function formatMapForTelemetry (map) {
-  // format from an object to a string map in order for
-  // telemetry intake to accept the configuration
-  return map
-    ? Object.entries(map).map(([key, value]) => `${key}:${value}`).join(',')
-    : ''
-}
-
-function updateConfig (changes, config) {
-  if (!config.telemetry.enabled) return
-  if (changes.length === 0) return
-
-  logger.trace(changes)
-
-  const application = createAppObject(config)
-  const host = createHostObject()
-
-  const nameMapping = {
-    sampleRate: 'DD_TRACE_SAMPLE_RATE',
-    logInjection: 'DD_LOG_INJECTION',
-    headerTags: 'DD_TRACE_HEADER_TAGS',
-    tags: 'DD_TAGS',
-    'sampler.rules': 'DD_TRACE_SAMPLING_RULES',
-    traceEnabled: 'DD_TRACE_ENABLED',
-    url: 'DD_TRACE_AGENT_URL',
-    'sampler.rateLimit': 'DD_TRACE_RATE_LIMIT',
-    queryStringObfuscation: 'DD_TRACE_OBFUSCATION_QUERY_STRING_REGEXP',
-    version: 'DD_VERSION',
-    env: 'DD_ENV',
-    service: 'DD_SERVICE',
-    clientIpHeader: 'DD_TRACE_CLIENT_IP_HEADER',
-    'grpc.client.error.statuses': 'DD_GRPC_CLIENT_ERROR_STATUSES',
-    'grpc.server.error.statuses': 'DD_GRPC_SERVER_ERROR_STATUSES'
-  }
+module.exports = {
+  start: (config, ...args) => {
+    if (!config?.telemetry?.enabled) return
 
-  const namesNeedFormatting = new Set(['DD_TAGS', 'peerServiceMapping', 'serviceMapping'])
+    telemetry = require('./telemetry')
+    telemetry.start(config, ...args)
+  },
 
-  const configuration = []
-  const names = [] // list of config names whose values have been changed
+  stop: () => {
+    if (telemetry === noop) return
 
-  for (const change of changes) {
-    const name = nameMapping[change.name] || change.name
+    telemetry.stop()
+    telemetry = noop
+  },
 
-    names.push(name)
-    const { origin, value } = change
-    const entry = { name, value, origin }
+  updateIntegrations: (...args) => telemetry.updateIntegrations(...args),
```

Comment:
I am uncertain why this is handled separately. Does removing this change anything?